### ISSUE TITLE
input weights

### ISSUE THREAD

**Issue by haiderasad** [shuaizzZ/Dice-Loss-PyTorch#1]
Hi, could you tell why the results are distorted? is it because I took weights as None

![image](https://user-images.githubusercontent.com/55244143/121467642-dc397400-c9d2-11eb-81a5-00ddb190ec5a.png)

**Comment by shuaizzZ** [comment]
Hi, the dice loss is discontinuous, just like the L1 loss will distort the result. It is suggested that a combination of dice and cross entropy loss may be better.

**Comment by haiderasad** [comment]
Ok meaning i correctly implemented the dice loss yes?

**Comment by shuaizzZ** [comment]
I think this is correct. I used the combined loss of Dice+CE and achieved better results than CE.

**Comment by haiderasad** [comment]
ok are you talking about exponential logarithmic loss or Combo loss kindly tell as both uses combination of dice plus CE

**Comment by shuaizzZ** [comment]
simply, 0.5*dice + 0.5*ce

**Comment by haiderasad** [comment]
Thank you much appreciate it

**Comment by haiderasad** [comment]
can you please tell why we multiply 0.5 with them?

**Comment by shuaizzZ** [comment]
0.5 is just an experience value, you can adjust the weight to suit your task.

**Comment by haiderasad** [comment]
my weight is None , so can i just add Dice +CE?

**Comment by shuaizzZ** [comment]
This is fine, but you need to adjust your lr accordingly, because diceloss+celoss> celoss

**Comment by haiderasad** [comment]
oh ok, yeah  i read that sometimes crossentropy can be greater than 1 so we have to somehow normalize it , they suggested dice+0.7CE , so i guess its just a matter of experimenting

**Comment by haiderasad** [comment]
And thank you so much for suggesting Dice+CE loss the results are much better now at 10000 iteration model, will train it further till 40000 as my GAN requires

![image](https://user-images.githubusercontent.com/55244143/122398108-68e1b600-cf2e-11eb-86a0-0cff32a456fb.png)

**Comment by shuaizzZ** [comment]
It seems that there is some distortion, you may need to pay attention to the damage of dice to small objects.

**Comment by shuaizzZ** [comment]
You can gradually reduce the weight of dice in the later stages of the training phase.

**Comment by haiderasad** [comment]
yes thats my main problem statement that the small objects gets overshadowed by the larger objects , so first i had implemented only CE and seeing the smaller objects not getting segmented i moved to dice loss.

Can you tell how to gradually reduce weight of dice?

**Comment by shuaizzZ** [comment]
For example, according to epoch or iteration, you can gradually reduce the weight of dice starting from 25000 iterations.
